inductiva/templating/manager.py

```diff
@@ -1,153 +1,39 @@
-                "templates": [f for f in files if is_template(f)],
-                "files": [f for f in files if not is_template(f)],
-            } for root, _, files in os.walk(self.template_dir, topdown=True)
-        }
-
-    def render_file(self,
-                    source_file: types.PathOrStr,
-                    target_file: Optional[types.PathOrStr] = None,
-                    overwrite: bool = False,
-                    **render_args) -> pathlib.Path:
-        """Render a single template to a file.
-
-        Renders the `source_file` from the template directory to `target_file`
-        inside the root directory, using the given render arguments.
-        If `target_file` is not given, the name of `source_file` is used and the
-        file is rendered directly to the root directory. When `overwrite` is
-        False (default) and the destination files already exists, a
-        FileExistsError will be raised.
-
-        Args:
-            source_file (str): The relative path of the template file inside
-                the template directory.
-            target_file: The name of the destination file. If not given, the
-                name of the template will be used and the rendered file will be
-                dumped to the root directory.
-            overwrite: If True, the destination file will be overwritten if it
-                already exists. If False (default), a FileExistsError will be
-                raised if the destination file already exists.
-            render_args: Keyword arguments that will be passed to the template
-                renderer.
-
-        Raises:
-            FileExistsError: If the destination file already exists and
-                `overwrite` is False.
-        """
-
-        if not self.renderer.is_template(source_file):
-            raise ValueError(f"{source_file} is not a template file.")
-
-        if target_file is None:
-            target_file = pathlib.Path(source_file).name
-        target_file = pathlib.Path(target_file)
-
-        target_file = self.renderer.strip_extension(target_file)
-        target_file = self.get_root_dir() / target_file
-
-        if not overwrite and target_file.exists():
-            raise FileExistsError(f"Target file {target_file} already exists.")
-
-        target_file.parent.mkdir(parents=True, exist_ok=True)
-        self.renderer.render(template_name=source_file,
-                             target_file=target_file,
-                             **render_args)
-
-        return target_file
-
-    def render_dir(self,
-                   source_dir: Optional[types.PathOrStr] = None,
-                   target_dir: Optional[types.PathOrStr] = None,
+    @classmethod
+    def render_dir(cls,
+                   source_dir: types.PathOrStr,
+                   target_dir: types.PathOrStr,
                    overwrite: bool = False,
-                   **render_args) -> pathlib.Path:
+                   **render_args):
```

Comment:
We are missing the return type in the docstring.

Reply:
Oh... thanks, good catch. But actually now this method does not need to return anything, as the target_dir is one of the inputs and it is where the data will appear (no internal logic to create a new directory with a different name)